{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# ngly1-ntwk\n",
    "\n",
    "## Purpose: \n",
    "* explore NGLY1 network, assembled by Nuria (~5,000 nodes, ~40,000 edges)\n",
    "* figure out how to best represent / store network data in python\n",
    "* calculate connections between nodes (node pathways)\n",
    "    * filter data\n",
    "    * aggregate / summarize pathways\n",
    "    * calc connections between terms\n",
    "* test out ways to visualize network\n",
    "\n",
    "### Authorship notes\n",
    "Laura Hughes, lhughes@scripps.edu\n",
    "\n",
    "based off of data assembled by Nuria Queralt Rosinach\n",
    "\n",
    "January 2018"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib as mp\n",
    "import string\n",
    "from collections import OrderedDict, defaultdict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id:ID</th>\n",
       "      <th>:LABEL</th>\n",
       "      <th>preflabel</th>\n",
       "      <th>synonyms:IGNORE</th>\n",
       "      <th>description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ZFIN:ZDB-GENE-051023-7</td>\n",
       "      <td>GENE</td>\n",
       "      <td>uba52</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>UBERON:0002107</td>\n",
       "      <td>ANAT</td>\n",
       "      <td>liver</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MGI:106594</td>\n",
       "      <td>GENE</td>\n",
       "      <td>Gk</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>UBERON:0004720</td>\n",
       "      <td>ANAT</td>\n",
       "      <td>cerebellar vermis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FlyBase:FBgn0033807</td>\n",
       "      <td>GENE</td>\n",
       "      <td>AQP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    id:ID :LABEL          preflabel synonyms:IGNORE  \\\n",
       "0  ZFIN:ZDB-GENE-051023-7   GENE              uba52             NaN   \n",
       "1          UBERON:0002107   ANAT              liver             NaN   \n",
       "2              MGI:106594   GENE                 Gk             NaN   \n",
       "3          UBERON:0004720   ANAT  cerebellar vermis             NaN   \n",
       "4     FlyBase:FBgn0033807   GENE                AQP             NaN   \n",
       "\n",
       "  description  \n",
       "0        None  \n",
       "1        None  \n",
       "2        None  \n",
       "3        None  \n",
       "4        None  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nodes = pd.read_csv('~/GitHub/ngly1/neo4j-community-3.0.3/import/ngly1/ngly1_concepts.tsv')\n",
    "nodes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>:START_ID</th>\n",
       "      <th>:TYPE</th>\n",
       "      <th>:END_ID</th>\n",
       "      <th>reference_uri</th>\n",
       "      <th>reference_supporting_text</th>\n",
       "      <th>reference_date</th>\n",
       "      <th>property_label</th>\n",
       "      <th>property_description</th>\n",
       "      <th>property_uri</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ZFIN:ZDB-GENE-051023-7</td>\n",
       "      <td>RO:0002206</td>\n",
       "      <td>UBERON:0002107</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>expressed in</td>\n",
       "      <td>None</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MGI:106594</td>\n",
       "      <td>RO:0002206</td>\n",
       "      <td>UBERON:0004720</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>expressed in</td>\n",
       "      <td>None</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FlyBase:FBgn0033807</td>\n",
       "      <td>RO:0002200</td>\n",
       "      <td>FBcv:0000349</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/20220848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>None</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_0002200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NCBIGene:824852</td>\n",
       "      <td>RO:HOM0000017</td>\n",
       "      <td>FlyBase:FBgn0033010</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>in orthology relationship with</td>\n",
       "      <td>None</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NCBIGene:470634</td>\n",
       "      <td>RO:HOM0000020</td>\n",
       "      <td>MGI:1342541</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>in 1 to 1 orthology relationship with</td>\n",
       "      <td>None</td>\n",
       "      <td>http://purl.obolibrary.org/obo/RO_HOM0000020</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                :START_ID          :TYPE              :END_ID  \\\n",
       "0  ZFIN:ZDB-GENE-051023-7     RO:0002206       UBERON:0002107   \n",
       "1              MGI:106594     RO:0002206       UBERON:0004720   \n",
       "2     FlyBase:FBgn0033807     RO:0002200         FBcv:0000349   \n",
       "3         NCBIGene:824852  RO:HOM0000017  FlyBase:FBgn0033010   \n",
       "4         NCBIGene:470634  RO:HOM0000020          MGI:1342541   \n",
       "\n",
       "                                  reference_uri  reference_supporting_text  \\\n",
       "0                                           NaN                        NaN   \n",
       "1                                           NaN                        NaN   \n",
       "2  https://www.ncbi.nlm.nih.gov/pubmed/20220848                        NaN   \n",
       "3                                           NaN                        NaN   \n",
       "4                                           NaN                        NaN   \n",
       "\n",
       "   reference_date                         property_label property_description  \\\n",
       "0             NaN                           expressed in                 None   \n",
       "1             NaN                           expressed in                 None   \n",
       "2             NaN                          has phenotype                 None   \n",
       "3             NaN         in orthology relationship with                 None   \n",
       "4             NaN  in 1 to 1 orthology relationship with                 None   \n",
       "\n",
       "                                   property_uri  \n",
       "0     http://purl.obolibrary.org/obo/RO_0002206  \n",
       "1     http://purl.obolibrary.org/obo/RO_0002206  \n",
       "2     http://purl.obolibrary.org/obo/RO_0002200  \n",
       "3  http://purl.obolibrary.org/obo/RO_HOM0000017  \n",
       "4  http://purl.obolibrary.org/obo/RO_HOM0000020  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "edges = pd.read_csv('~/GitHub/ngly1/neo4j-community-3.0.3/import/ngly1/ngly1_statements.tsv')\n",
    "edges.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>:START_ID</th>\n",
       "      <th>:TYPE</th>\n",
       "      <th>:END_ID</th>\n",
       "      <th>reference_uri</th>\n",
       "      <th>reference_supporting_text</th>\n",
       "      <th>reference_date</th>\n",
       "      <th>property_description</th>\n",
       "      <th>property_uri</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>property_label</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>None</th>\n",
       "      <td>1274</td>\n",
       "      <td>1274</td>\n",
       "      <td>1274</td>\n",
       "      <td>1011</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1274</td>\n",
       "      <td>1274</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Source</th>\n",
       "      <td>49</td>\n",
       "      <td>49</td>\n",
       "      <td>49</td>\n",
       "      <td>49</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "      <td>49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>colocalizes with</th>\n",
       "      <td>60</td>\n",
       "      <td>60</td>\n",
       "      <td>60</td>\n",
       "      <td>60</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>60</td>\n",
       "      <td>60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>contributes to</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>derives_from</th>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>enables</th>\n",
       "      <td>582</td>\n",
       "      <td>582</td>\n",
       "      <td>582</td>\n",
       "      <td>582</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>582</td>\n",
       "      <td>582</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>exact match</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>expressed in</th>\n",
       "      <td>2303</td>\n",
       "      <td>2303</td>\n",
       "      <td>2303</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2303</td>\n",
       "      <td>2303</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genetically interacts with</th>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>has gene product</th>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>has phenotype</th>\n",
       "      <td>6246</td>\n",
       "      <td>6246</td>\n",
       "      <td>6246</td>\n",
       "      <td>5921</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6244</td>\n",
       "      <td>6244</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>has role</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>has_affected_feature</th>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>has_genotype</th>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>in 1 to 1 orthology relationship with</th>\n",
       "      <td>8854</td>\n",
       "      <td>8854</td>\n",
       "      <td>8854</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>8854</td>\n",
       "      <td>8854</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>in orthology relationship with</th>\n",
       "      <td>15207</td>\n",
       "      <td>15207</td>\n",
       "      <td>15207</td>\n",
       "      <td>263</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15206</td>\n",
       "      <td>15206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>in paralogy relationship with</th>\n",
       "      <td>2568</td>\n",
       "      <td>2568</td>\n",
       "      <td>2568</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2568</td>\n",
       "      <td>2568</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>instance of</th>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>interacts with</th>\n",
       "      <td>2885</td>\n",
       "      <td>2885</td>\n",
       "      <td>2885</td>\n",
       "      <td>903</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2858</td>\n",
       "      <td>2858</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>involved in</th>\n",
       "      <td>779</td>\n",
       "      <td>779</td>\n",
       "      <td>779</td>\n",
       "      <td>531</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>762</td>\n",
       "      <td>762</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>is marker for</th>\n",
       "      <td>119</td>\n",
       "      <td>119</td>\n",
       "      <td>119</td>\n",
       "      <td>119</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>119</td>\n",
       "      <td>119</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>is model of</th>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>is part of</th>\n",
       "      <td>1043</td>\n",
       "      <td>1043</td>\n",
       "      <td>1043</td>\n",
       "      <td>1031</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1031</td>\n",
       "      <td>1031</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>is_allele_of</th>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pathogenic_for_condition</th>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>see also</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       :START_ID  :TYPE  :END_ID  \\\n",
       "property_label                                                     \n",
       "None                                        1274   1274     1274   \n",
       "Source                                        49     49       49   \n",
       "colocalizes with                              60     60       60   \n",
       "contributes to                                 1      1        1   \n",
       "derives_from                                   9      9        9   \n",
       "enables                                      582    582      582   \n",
       "exact match                                    2      2        2   \n",
       "expressed in                                2303   2303     2303   \n",
       "genetically interacts with                    12     12       12   \n",
       "has gene product                               7      7        7   \n",
       "has phenotype                               6246   6246     6246   \n",
       "has role                                       2      2        2   \n",
       "has_affected_feature                          30     30       30   \n",
       "has_genotype                                  21     21       21   \n",
       "in 1 to 1 orthology relationship with       8854   8854     8854   \n",
       "in orthology relationship with             15207  15207    15207   \n",
       "in paralogy relationship with               2568   2568     2568   \n",
       "instance of                                    7      7        7   \n",
       "interacts with                              2885   2885     2885   \n",
       "involved in                                  779    779      779   \n",
       "is marker for                                119    119      119   \n",
       "is model of                                   17     17       17   \n",
       "is part of                                  1043   1043     1043   \n",
       "is_allele_of                                  18     18       18   \n",
       "pathogenic_for_condition                      15     15       15   \n",
       "see also                                       1      1        1   \n",
       "\n",
       "                                       reference_uri  \\\n",
       "property_label                                         \n",
       "None                                            1011   \n",
       "Source                                            49   \n",
       "colocalizes with                                  60   \n",
       "contributes to                                     1   \n",
       "derives_from                                       0   \n",
       "enables                                          582   \n",
       "exact match                                        0   \n",
       "expressed in                                       0   \n",
       "genetically interacts with                        12   \n",
       "has gene product                                   0   \n",
       "has phenotype                                   5921   \n",
       "has role                                           0   \n",
       "has_affected_feature                               0   \n",
       "has_genotype                                       0   \n",
       "in 1 to 1 orthology relationship with              1   \n",
       "in orthology relationship with                   263   \n",
       "in paralogy relationship with                      0   \n",
       "instance of                                        0   \n",
       "interacts with                                   903   \n",
       "involved in                                      531   \n",
       "is marker for                                    119   \n",
       "is model of                                        1   \n",
       "is part of                                      1031   \n",
       "is_allele_of                                       0   \n",
       "pathogenic_for_condition                          15   \n",
       "see also                                           0   \n",
       "\n",
       "                                       reference_supporting_text  \\\n",
       "property_label                                                     \n",
       "None                                                           0   \n",
       "Source                                                         0   \n",
       "colocalizes with                                               0   \n",
       "contributes to                                                 0   \n",
       "derives_from                                                   0   \n",
       "enables                                                        0   \n",
       "exact match                                                    0   \n",
       "expressed in                                                   0   \n",
       "genetically interacts with                                     0   \n",
       "has gene product                                               0   \n",
       "has phenotype                                                  0   \n",
       "has role                                                       0   \n",
       "has_affected_feature                                           0   \n",
       "has_genotype                                                   0   \n",
       "in 1 to 1 orthology relationship with                          0   \n",
       "in orthology relationship with                                 0   \n",
       "in paralogy relationship with                                  0   \n",
       "instance of                                                    0   \n",
       "interacts with                                                 0   \n",
       "involved in                                                    0   \n",
       "is marker for                                                  0   \n",
       "is model of                                                    0   \n",
       "is part of                                                     0   \n",
       "is_allele_of                                                   0   \n",
       "pathogenic_for_condition                                       0   \n",
       "see also                                                       0   \n",
       "\n",
       "                                       reference_date  property_description  \\\n",
       "property_label                                                                \n",
       "None                                                0                  1274   \n",
       "Source                                              0                    49   \n",
       "colocalizes with                                    0                    60   \n",
       "contributes to                                      0                     1   \n",
       "derives_from                                        0                     9   \n",
       "enables                                             0                   582   \n",
       "exact match                                         0                     0   \n",
       "expressed in                                        0                  2303   \n",
       "genetically interacts with                          0                    12   \n",
       "has gene product                                    0                     0   \n",
       "has phenotype                                       0                  6244   \n",
       "has role                                            0                     0   \n",
       "has_affected_feature                                0                    30   \n",
       "has_genotype                                        0                    21   \n",
       "in 1 to 1 orthology relationship with               0                  8854   \n",
       "in orthology relationship with                      0                 15206   \n",
       "in paralogy relationship with                       0                  2568   \n",
       "instance of                                         0                     0   \n",
       "interacts with                                      0                  2858   \n",
       "involved in                                         0                   762   \n",
       "is marker for                                       0                   119   \n",
       "is model of                                         0                    17   \n",
       "is part of                                          0                  1031   \n",
       "is_allele_of                                        0                    18   \n",
       "pathogenic_for_condition                            0                    15   \n",
       "see also                                            0                     0   \n",
       "\n",
       "                                       property_uri  \n",
       "property_label                                       \n",
       "None                                           1274  \n",
       "Source                                           49  \n",
       "colocalizes with                                 60  \n",
       "contributes to                                    1  \n",
       "derives_from                                      9  \n",
       "enables                                         582  \n",
       "exact match                                       0  \n",
       "expressed in                                   2303  \n",
       "genetically interacts with                       12  \n",
       "has gene product                                  0  \n",
       "has phenotype                                  6244  \n",
       "has role                                          0  \n",
       "has_affected_feature                             30  \n",
       "has_genotype                                     21  \n",
       "in 1 to 1 orthology relationship with          8854  \n",
       "in orthology relationship with                15206  \n",
       "in paralogy relationship with                  2568  \n",
       "instance of                                       0  \n",
       "interacts with                                 2858  \n",
       "involved in                                     762  \n",
       "is marker for                                   119  \n",
       "is model of                                      17  \n",
       "is part of                                     1031  \n",
       "is_allele_of                                     18  \n",
       "pathogenic_for_condition                         15  \n",
       "see also                                          0  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "edges.groupby(['property_label']).count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>:START_ID</th>\n",
       "      <th>:END_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>25373</th>\n",
       "      <td>:.well-known/genid/H_AQP1_C01_TM1B</td>\n",
       "      <td>MGI:103201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14343</th>\n",
       "      <td>:.well-known/genid/H_AQP1_C01_TM1B</td>\n",
       "      <td>MGI:5692545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30776</th>\n",
       "      <td>:.well-known/genid/MGI101922-VLU-MGI102724-VLU...</td>\n",
       "      <td>MGI:103201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17460</th>\n",
       "      <td>:.well-known/genid/MGI101922-VLU-MGI102724-VLU...</td>\n",
       "      <td>MGI:1100851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22606</th>\n",
       "      <td>:.well-known/genid/MGI102702-VLU-MGI103201-VLU...</td>\n",
       "      <td>MGI:1338871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12601</th>\n",
       "      <td>:.well-known/genid/MGI102702-VLU-MGI103201-VLU...</td>\n",
       "      <td>MGI:103201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35552</th>\n",
       "      <td>:.well-known/genid/MGI103076-VLU-MGI103198-VLU...</td>\n",
       "      <td>MGI:103201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32547</th>\n",
       "      <td>:.well-known/genid/MGI103201-VL</td>\n",
       "      <td>MGI:103201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19824</th>\n",
       "      <td>:.well-known/genid/MGI103201-VLU-MGI107387-VLU...</td>\n",
       "      <td>MGI:107387</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23297</th>\n",
       "      <td>:.well-known/genid/MGI103201-VLU-MGI107387-VLU...</td>\n",
       "      <td>MGI:103201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8180</th>\n",
       "      <td>:.well-known/genid/MGI5692545GENO0000137</td>\n",
       "      <td>MGI:103201</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               :START_ID      :END_ID\n",
       "25373                 :.well-known/genid/H_AQP1_C01_TM1B   MGI:103201\n",
       "14343                 :.well-known/genid/H_AQP1_C01_TM1B  MGI:5692545\n",
       "30776  :.well-known/genid/MGI101922-VLU-MGI102724-VLU...   MGI:103201\n",
       "17460  :.well-known/genid/MGI101922-VLU-MGI102724-VLU...  MGI:1100851\n",
       "22606  :.well-known/genid/MGI102702-VLU-MGI103201-VLU...  MGI:1338871\n",
       "12601  :.well-known/genid/MGI102702-VLU-MGI103201-VLU...   MGI:103201\n",
       "35552  :.well-known/genid/MGI103076-VLU-MGI103198-VLU...   MGI:103201\n",
       "32547                    :.well-known/genid/MGI103201-VL   MGI:103201\n",
       "19824  :.well-known/genid/MGI103201-VLU-MGI107387-VLU...   MGI:107387\n",
       "23297  :.well-known/genid/MGI103201-VLU-MGI107387-VLU...   MGI:103201\n",
       "8180            :.well-known/genid/MGI5692545GENO0000137   MGI:103201"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "edges2 = edges.sort_values(':START_ID')\n",
    "\n",
    "edges2.head()\n",
    "\n",
    "edges2 = edges2[0:11][[':START_ID', ':END_ID']]\n",
    "edges2.set_index(':START_ID')\n",
    "\n",
    "edges2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{':END_ID': {8180: 'MGI:103201',\n",
       "  12601: 'MGI:103201',\n",
       "  14343: 'MGI:5692545',\n",
       "  17460: 'MGI:1100851',\n",
       "  19824: 'MGI:107387',\n",
       "  22606: 'MGI:1338871',\n",
       "  23297: 'MGI:103201',\n",
       "  25373: 'MGI:103201',\n",
       "  30776: 'MGI:103201',\n",
       "  32547: 'MGI:103201',\n",
       "  35552: 'MGI:103201'},\n",
       " ':START_ID': {8180: ':.well-known/genid/MGI5692545GENO0000137',\n",
       "  12601: ':.well-known/genid/MGI102702-VLU-MGI103201-VLU-MGI104576-VLU-MGI104697-VLU-MGI104960-VLU-MGI105370-VLU-MGI105959-VLU-MGI107560-VLU-MGI107690-VLU-MGI107726-VLU-MGI108559-VLU-MGI1095407-VLU-MGI1096573-VLU-MGI1098550-VLU-MGI1196391-VLU-MGI1276116-VLU-MGI1277238-VLU-MGI1316658-VLU-MGI1330860-VLU-MGI1336167-VLU-MGI1338871-VLU-MGI1342290-VLU-MGI1343180-VLU-MGI1353576-VLU-MGI1858414-VLU-MGI1889341-VLU-MGI1891717-VLU-MGI1891830-VLU-MGI1913429-VLU-MGI1913855-VLU-MGI1914080-VLU-MGI1914269-VLU-MGI1914324-VLU-MGI1915134-VLU-MGI1915467-VLU-MGI1917452-VLU-MGI1917695-VLU-MGI1918180-VLU-MGI1918195-VLU-MGI1918940-VLU-MGI1919174-VLU-MGI1919318-VLU-MGI1919429-VLU-MGI1919830-VLU-MGI1921337-VLU-MGI1921426-VLU-MGI1922567-VLU-MGI1923017-VLU-MGI1923691-VLU-MGI1924303-VLU-MGI1928760-VLU-MGI1929237-VLU-MGI1931256-VLU-MGI1933210-VLU-MGI1933532-VLU-MGI2139806-VLU-MGI2178429-VLU-MGI2183752-VLU-MGI2385848-VLU-MGI2442032-VLU-MGI2443052-VLU-MGI2443331-VLU-MGI2444148-VLU-MGI2445210-VLU-MGI2447811-VLU-MGI2667778-VLU-MGI2675387-VLU-MGI2681865-VLU-MGI2684953-VLU-MGI2685175-VLU-MGI3030601-VLU-MGI3030938-VLU-MGI3588222-VLU-MGI4834570-VLU-MGI88075-VLU-MGI88353-VLU-MGI894323-VLU-MGI95931-VLU-MGI96955-VLU-MGI98881-VLU-GENO0000649-MMRRC038430',\n",
       "  14343: ':.well-known/genid/H_AQP1_C01_TM1B',\n",
       "  17460: ':.well-known/genid/MGI101922-VLU-MGI102724-VLU-MGI103201-VLU-MGI104510-VLU-MGI104517-VLU-MGI104785-VLU-MGI105313-VLU-MGI106562-VLU-MGI107437-VLU-MGI107624-VLU-MGI107732-VLU-MGI108391-VLU-MGI109280-VLU-MGI109311-VLU-MGI109351-VLU-MGI109367-VLU-MGI1096383-VLU-MGI1098754-VLU-MGI1098815-VLU-MGI1100851-VLU-MGI1194898-VLU-MGI1196326-VLU-MGI1203730-VLU-MGI1277977-VLU-MGI1298386-VLU-MGI1316648-VLU-MGI1316652-VLU-MGI1333752-VLU-MGI1333884-VLU-MGI1338824-VLU-MGI1341110-VLU-MGI1341823-VLU-MGI1345184-VLU-MGI1346021-VLU-MGI1349385-VLU-MGI1351617-VLU-MGI1354756-VLU-MGI1858225-VLU-MGI1858232-VLU-MGI1889802-VLU-MGI1890647-VLU-MGI1913704-VLU-MGI1914171-VLU-MGI1914404-VLU-MGI1914582-VLU-MGI1914740-VLU-MGI1915207-VLU-MGI1915293-VLU-MGI1915367-VLU-MGI1917097-VLU-MGI1918560-VLU-MGI1918780-VLU-MGI1918885-VLU-MGI1919489-VLU-MGI1919563-VLU-MGI1919735-VLU-MGI1920175-VLU-MGI1921370-VLU-MGI1921430-VLU-MGI1921620-VLU-MGI1924662-VLU-MGI1926033-VLU-MGI1926048-VLU-MGI1926163-VLU-MGI1926803-VLU-MGI1930140-VLU-MGI1933786-VLU-MGI1935219-VLU-MGI2139607-VLU-MGI2143886-VLU-MGI2148793-VLU-MGI2153525-VLU-MGI2178742-VLU-MGI2178743-VLU-MGI2384495-VLU-MGI2384786-VLU-MGI2385848-VLU-MGI2389007-VLU-MGI2429859-VLU-MGI2441745-VLU-MGI2442071-VLU-MGI2442221-VLU-MGI2442728-VLU-MGI2447024-VLU-MGI2450529-VLU-MGI2679005-VLU-MGI2679684-VLU-MGI2684899-VLU-MGI2684923-VLU-MGI2685289-VLU-MGI2685899-VLU-MGI3030934-VLU-MGI3031342-VLU-MGI3525150-VLU-MGI3644875-VLU-MGI3645902-VLU-MGI3717143-VLU-MGI88016-VLU-MGI88052-VLU-MGI88602-VLU-MGI95392-VLU-MGI95558-VLU-MGI96010-VLU-MGI96161-VLU-MGI96575-VLU-MGI97501-VLU-MGI97797-VLU-MGI98259-VLU-MGI98733-VLU-MGI98926-VLU-MGI99400-VLU-MGI99448-VLU-GENO0000649-MMRRC042662',\n",
       "  19824: ':.well-known/genid/MGI103201-VLU-MGI107387-VLU-MGI108391-VLU-MGI108554-VLU-MGI109234-VLU-MGI109304-VLU-MGI109536-VLU-MGI109548-VLU-MGI1096345-VLU-MGI1098533-VLU-MGI1202302-VLU-MGI1202397-VLU-MGI1298369-VLU-MGI1315213-VLU-MGI1341839-VLU-MGI1341847-VLU-MGI1347346-VLU-MGI1858313-VLU-MGI1858330-VLU-MGI1860197-VLU-MGI1862037-VLU-MGI1913496-VLU-MGI1913837-VLU-MGI1914689-VLU-MGI1914742-VLU-MGI1915760-VLU-MGI1915851-VLU-MGI1917193-VLU-MGI1918135-VLU-MGI1918659-VLU-MGI1918772-VLU-MGI1919019-VLU-MGI1921500-VLU-MGI1922664-VLU-MGI1922790-VLU-MGI1923665-VLU-MGI1923935-VLU-MGI1924140-VLU-MGI1929289-VLU-MGI1932389-VLU-MGI2135601-VLU-MGI2142195-VLU-MGI2442266-VLU-MGI2442600-VLU-MGI2443417-VLU-MGI2443713-VLU-MGI2444047-VLU-MGI2444178-VLU-MGI2444371-VLU-MGI2652818-VLU-MGI2673874-VLU-MGI2682211-VLU-MGI2685287-VLU-MGI2685483-VLU-MGI2685720-VLU-MGI3030137-VLU-MGI3030199-VLU-MGI3030372-VLU-MGI3030587-VLU-MGI3031053-VLU-MGI3045962-VLU-MGI3606242-VLU-MGI3615275-VLU-MGI3648937-VLU-MGI3652306-VLU-MGI5477151-VLU-MGI88382-VLU-MGI88452-VLU-MGI894810-VLU-MGI95480-VLU-MGI95489-VLU-MGI95514-VLU-MGI95752-VLU-MGI96770-VLU-GENO0000649-MMRRC041819',\n",
       "  22606: ':.well-known/genid/MGI102702-VLU-MGI103201-VLU-MGI104576-VLU-MGI104697-VLU-MGI104960-VLU-MGI105370-VLU-MGI105959-VLU-MGI107560-VLU-MGI107690-VLU-MGI107726-VLU-MGI108559-VLU-MGI1095407-VLU-MGI1096573-VLU-MGI1098550-VLU-MGI1196391-VLU-MGI1276116-VLU-MGI1277238-VLU-MGI1316658-VLU-MGI1330860-VLU-MGI1336167-VLU-MGI1338871-VLU-MGI1342290-VLU-MGI1343180-VLU-MGI1353576-VLU-MGI1858414-VLU-MGI1889341-VLU-MGI1891717-VLU-MGI1891830-VLU-MGI1913429-VLU-MGI1913855-VLU-MGI1914080-VLU-MGI1914269-VLU-MGI1914324-VLU-MGI1915134-VLU-MGI1915467-VLU-MGI1917452-VLU-MGI1917695-VLU-MGI1918180-VLU-MGI1918195-VLU-MGI1918940-VLU-MGI1919174-VLU-MGI1919318-VLU-MGI1919429-VLU-MGI1919830-VLU-MGI1921337-VLU-MGI1921426-VLU-MGI1922567-VLU-MGI1923017-VLU-MGI1923691-VLU-MGI1924303-VLU-MGI1928760-VLU-MGI1929237-VLU-MGI1931256-VLU-MGI1933210-VLU-MGI1933532-VLU-MGI2139806-VLU-MGI2178429-VLU-MGI2183752-VLU-MGI2385848-VLU-MGI2442032-VLU-MGI2443052-VLU-MGI2443331-VLU-MGI2444148-VLU-MGI2445210-VLU-MGI2447811-VLU-MGI2667778-VLU-MGI2675387-VLU-MGI2681865-VLU-MGI2684953-VLU-MGI2685175-VLU-MGI3030601-VLU-MGI3030938-VLU-MGI3588222-VLU-MGI4834570-VLU-MGI88075-VLU-MGI88353-VLU-MGI894323-VLU-MGI95931-VLU-MGI96955-VLU-MGI98881-VLU-GENO0000649-MMRRC038430',\n",
       "  23297: ':.well-known/genid/MGI103201-VLU-MGI107387-VLU-MGI108391-VLU-MGI108554-VLU-MGI109234-VLU-MGI109304-VLU-MGI109536-VLU-MGI109548-VLU-MGI1096345-VLU-MGI1098533-VLU-MGI1202302-VLU-MGI1202397-VLU-MGI1298369-VLU-MGI1315213-VLU-MGI1341839-VLU-MGI1341847-VLU-MGI1347346-VLU-MGI1858313-VLU-MGI1858330-VLU-MGI1860197-VLU-MGI1862037-VLU-MGI1913496-VLU-MGI1913837-VLU-MGI1914689-VLU-MGI1914742-VLU-MGI1915760-VLU-MGI1915851-VLU-MGI1917193-VLU-MGI1918135-VLU-MGI1918659-VLU-MGI1918772-VLU-MGI1919019-VLU-MGI1921500-VLU-MGI1922664-VLU-MGI1922790-VLU-MGI1923665-VLU-MGI1923935-VLU-MGI1924140-VLU-MGI1929289-VLU-MGI1932389-VLU-MGI2135601-VLU-MGI2142195-VLU-MGI2442266-VLU-MGI2442600-VLU-MGI2443417-VLU-MGI2443713-VLU-MGI2444047-VLU-MGI2444178-VLU-MGI2444371-VLU-MGI2652818-VLU-MGI2673874-VLU-MGI2682211-VLU-MGI2685287-VLU-MGI2685483-VLU-MGI2685720-VLU-MGI3030137-VLU-MGI3030199-VLU-MGI3030372-VLU-MGI3030587-VLU-MGI3031053-VLU-MGI3045962-VLU-MGI3606242-VLU-MGI3615275-VLU-MGI3648937-VLU-MGI3652306-VLU-MGI5477151-VLU-MGI88382-VLU-MGI88452-VLU-MGI894810-VLU-MGI95480-VLU-MGI95489-VLU-MGI95514-VLU-MGI95752-VLU-MGI96770-VLU-GENO0000649-MMRRC041819',\n",
       "  25373: ':.well-known/genid/H_AQP1_C01_TM1B',\n",
       "  30776: ':.well-known/genid/MGI101922-VLU-MGI102724-VLU-MGI103201-VLU-MGI104510-VLU-MGI104517-VLU-MGI104785-VLU-MGI105313-VLU-MGI106562-VLU-MGI107437-VLU-MGI107624-VLU-MGI107732-VLU-MGI108391-VLU-MGI109280-VLU-MGI109311-VLU-MGI109351-VLU-MGI109367-VLU-MGI1096383-VLU-MGI1098754-VLU-MGI1098815-VLU-MGI1100851-VLU-MGI1194898-VLU-MGI1196326-VLU-MGI1203730-VLU-MGI1277977-VLU-MGI1298386-VLU-MGI1316648-VLU-MGI1316652-VLU-MGI1333752-VLU-MGI1333884-VLU-MGI1338824-VLU-MGI1341110-VLU-MGI1341823-VLU-MGI1345184-VLU-MGI1346021-VLU-MGI1349385-VLU-MGI1351617-VLU-MGI1354756-VLU-MGI1858225-VLU-MGI1858232-VLU-MGI1889802-VLU-MGI1890647-VLU-MGI1913704-VLU-MGI1914171-VLU-MGI1914404-VLU-MGI1914582-VLU-MGI1914740-VLU-MGI1915207-VLU-MGI1915293-VLU-MGI1915367-VLU-MGI1917097-VLU-MGI1918560-VLU-MGI1918780-VLU-MGI1918885-VLU-MGI1919489-VLU-MGI1919563-VLU-MGI1919735-VLU-MGI1920175-VLU-MGI1921370-VLU-MGI1921430-VLU-MGI1921620-VLU-MGI1924662-VLU-MGI1926033-VLU-MGI1926048-VLU-MGI1926163-VLU-MGI1926803-VLU-MGI1930140-VLU-MGI1933786-VLU-MGI1935219-VLU-MGI2139607-VLU-MGI2143886-VLU-MGI2148793-VLU-MGI2153525-VLU-MGI2178742-VLU-MGI2178743-VLU-MGI2384495-VLU-MGI2384786-VLU-MGI2385848-VLU-MGI2389007-VLU-MGI2429859-VLU-MGI2441745-VLU-MGI2442071-VLU-MGI2442221-VLU-MGI2442728-VLU-MGI2447024-VLU-MGI2450529-VLU-MGI2679005-VLU-MGI2679684-VLU-MGI2684899-VLU-MGI2684923-VLU-MGI2685289-VLU-MGI2685899-VLU-MGI3030934-VLU-MGI3031342-VLU-MGI3525150-VLU-MGI3644875-VLU-MGI3645902-VLU-MGI3717143-VLU-MGI88016-VLU-MGI88052-VLU-MGI88602-VLU-MGI95392-VLU-MGI95558-VLU-MGI96010-VLU-MGI96161-VLU-MGI96575-VLU-MGI97501-VLU-MGI97797-VLU-MGI98259-VLU-MGI98733-VLU-MGI98926-VLU-MGI99400-VLU-MGI99448-VLU-GENO0000649-MMRRC042662',\n",
       "  32547: ':.well-known/genid/MGI103201-VL',\n",
       "  35552: ':.well-known/genid/MGI103076-VLU-MGI103198-VLU-MGI103201-VLU-MGI104627-VLU-MGI104779-VLU-MGI105976-VLU-MGI106018-VLU-MGI106645-VLU-MGI107732-VLU-MGI107996-VLU-MGI109351-VLU-MGI109569-VLU-MGI109618-VLU-MGI1098754-VLU-MGI1098815-VLU-MGI1100864-VLU-MGI1194898-VLU-MGI1196326-VLU-MGI1203730-VLU-MGI1289279-VLU-MGI1316652-VLU-MGI1339712-VLU-MGI1341110-VLU-MGI1345184-VLU-MGI1349385-VLU-MGI1349721-VLU-MGI1351617-VLU-MGI1858232-VLU-MGI1858304-VLU-MGI1859555-VLU-MGI1888526-VLU-MGI1889294-VLU-MGI1889802-VLU-MGI1890647-VLU-MGI1891361-VLU-MGI1914404-VLU-MGI1914582-VLU-MGI1915367-VLU-MGI1915851-VLU-MGI1916068-VLU-MGI1916394-VLU-MGI1917685-VLU-MGI1917728-VLU-MGI1918255-VLU-MGI1918263-VLU-MGI1918441-VLU-MGI1918780-VLU-MGI1918988-VLU-MGI1919074-VLU-MGI1919563-VLU-MGI1919936-VLU-MGI1921370-VLU-MGI1921430-VLU-MGI1921446-VLU-MGI1921612-VLU-MGI1924306-VLU-MGI1925505-VLU-MGI1926217-VLU-MGI1929211-VLU-MGI1933786-VLU-MGI2137383-VLU-MGI2137680-VLU-MGI2142048-VLU-MGI2143886-VLU-MGI2145955-VLU-MGI2146009-VLU-MGI2148793-VLU-MGI2153272-VLU-MGI2153482-VLU-MGI2177284-VLU-MGI2178743-VLU-MGI2384495-VLU-MGI2387634-VLU-MGI2441745-VLU-MGI2442119-VLU-MGI2442784-VLU-MGI2443016-VLU-MGI2444274-VLU-MGI2444430-VLU-MGI2444928-VLU-MGI2446120-VLU-MGI2447024-VLU-MGI2450529-VLU-MGI2450757-VLU-MGI2676311-VLU-MGI2684924-VLU-MGI2685289-VLU-MGI2685821-VLU-MGI3030144-VLU-MGI3030440-VLU-MGI3030502-VLU-MGI3030509-VLU-MGI3030597-VLU-MGI3030705-VLU-MGI3030934-VLU-MGI3031168-VLU-MGI3035181-VLU-MGI3616079-VLU-MGI3641753-VLU-MGI3644875-VLU-MGI3645984-VLU-MGI3646469-VLU-MGI3648316-VLU-MGI3648919-VLU-MGI5477020-VLU-MGI88016-VLU-MGI88052-VLU-MGI88602-VLU-MGI894702-VLU-MGI95392-VLU-MGI96010-VLU-MGI96235-VLU-MGI96605-VLU-MGI96929-VLU-MGI97479-VLU-MGI97770-VLU-MGI97797-VLU-MGI98730-VLU-MGI98926-VLU-MGI98935-VLU-MGI99400-VLU-MGI99448-VLU-GENO0000649-MMRRC042663'}}"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dd = defaultdict(list)\n",
    "df= edges2.to_dict()\n",
    "\n",
    "edges2\n",
    "df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['a',\n",
       " 'b',\n",
       " 'c',\n",
       " 'd',\n",
       " 'e',\n",
       " 'f',\n",
       " 'g',\n",
       " 'h',\n",
       " 'i',\n",
       " 'j',\n",
       " 'k',\n",
       " 'l',\n",
       " 'm',\n",
       " 'n',\n",
       " 'o',\n",
       " 'p',\n",
       " 'q',\n",
       " 'r',\n",
       " 's',\n",
       " 't',\n",
       " 'u',\n",
       " 'v',\n",
       " 'w',\n",
       " 'x',\n",
       " 'y',\n",
       " 'z',\n",
       " 'A',\n",
       " 'B',\n",
       " 'C',\n",
       " 'D',\n",
       " 'E',\n",
       " 'F',\n",
       " 'G',\n",
       " 'H',\n",
       " 'I',\n",
       " 'J',\n",
       " 'K',\n",
       " 'L',\n",
       " 'M',\n",
       " 'N',\n",
       " 'O',\n",
       " 'P',\n",
       " 'Q',\n",
       " 'R',\n",
       " 'S',\n",
       " 'T',\n",
       " 'U',\n",
       " 'V',\n",
       " 'W',\n",
       " 'X',\n",
       " 'Y',\n",
       " 'Z']"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(string.ascii_letters)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
